dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java

```diff
@@ -212,9 +228,26 @@ private String generateReferenceBeanName(AnnotationAttributes attributes, Class<
         if (!attributes.isEmpty()) {
             beanNameBuilder.append('(');
             for (Map.Entry<String, Object> entry : attributes.entrySet()) {
+                String value;
+                if (entry.getValue().getClass().isArray()) {
+                    String[] entryValues = (String[]) entry.getValue();
+                    if ("parameters".equals(entry.getKey())) {
+                        // parameters spec is {key1,value1,key2,value2}
+                        ArrayList<String> kvList = new ArrayList<>();
+                        for (int i = 0; i < entryValues.length / 2 * 2; i = i + 2) {
```

Comment:
` i < entryValues.length / 2 * 2` changed to  `i < entryValues.length`.

Reply:
It's for protect mistake useage like {key1,value1,key2,value2,key3}